src/main/File/File.java

```diff
@@ -0,0 +1,155 @@
+package File;
+
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import com.mongodb.client.gridfs.model.GridFSFile;
+import com.mongodb.client.model.Filters;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.bson.codecs.pojo.annotations.BsonIgnore;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+import org.bson.types.ObjectId;
+
+import java.io.InputStream;
+import java.util.Date;
+import java.util.Objects;
+
+@Slf4j
+public class File {
+  @Getter @Setter private ObjectId id; // id of file in collection (metadata)
+  @Getter @Setter private ObjectId fileId; // id of file in bucket (actual file bytes)
+  @Getter @Setter @BsonIgnore private InputStream fileStream;
+  @Getter @Setter private String filename;
+  @Getter @Setter private FileType fileType;
+  @Getter @Setter private Date uploadedAt;
+  @Getter @Setter private String username;
+  @Getter @Setter private String organizationName;
+
+  @Getter
+  @Setter
+  @BsonProperty(value = "annotated")
+  private boolean isAnnotated;
+
+  @Getter @Setter private String contentType;
+
+  public File() {}
+
+  public File(
+      ObjectId id,
+      String filename,
+      FileType fileType,
+      Date uploadedAt,
+      String username,
+      String organizationName,
+      String contentType) {
+    this.id = id;
+    this.filename = filename;
+    this.fileType = fileType;
+    this.uploadedAt = uploadedAt;
+    this.username = username;
+    this.organizationName = organizationName;
+    this.contentType = contentType;
+  }
+
+  public File(
+      String username,
+      Date uploadedAt,
+      InputStream fileStream,
+      FileType fileType,
+      String fileName,
+      String organizationName,
+      boolean isAnnotated) {
+    this.id = new ObjectId();
+    this.username = username;
+    this.uploadedAt = uploadedAt;
+    this.fileStream = fileStream;
+    this.fileType = fileType;
+    this.filename = fileName;
+    this.organizationName = organizationName;
+    this.isAnnotated = isAnnotated;
+  }
+
+  public File(
+      String username,
+      Date uploadedAt,
+      InputStream fileStream,
+      FileType fileType,
+      String fileName) {
+    this.id = new ObjectId();
+    this.username = username;
+    this.uploadedAt = uploadedAt;
+    this.fileStream = fileStream;
+    this.fileType = fileType;
+    this.filename = fileName;
+  }
+
+  public File(String username, Date uploadedAt, InputStream fileStream, FileType fileType) {
+    this.id = new ObjectId();
+    this.username = username;
+    this.uploadedAt = uploadedAt;
+    this.fileStream = fileStream;
+    this.fileType = fileType;
+  }
+
+  public File(String username, Date uploadedAt, ObjectId fileId, FileType fileType) {
+    this.id = new ObjectId();
```

Comment:
is this constructor necessary? I would prefer maybe just one or two constructors and we can just pass in Optional<> fields if we think those are not as applicable to every File